subsys/net/ip/tcp2.c

```diff
@@ -218,22 +222,29 @@ static const char *tcp_flags(uint8_t flags)
 	return buf;
 }
 
-static size_t tcp_data_len(struct net_pkt *pkt)
+static size_t tcp_data_len(struct net_pkt *pkt, size_t *headers_len)
```

Comment:
> Maybe `headers_len` should be renamed to `data_offset` so that this function returns both offset and length of data. In this case we still need a better name for this function.

Well, we return the data length as earlier (thus I kept the function name). Also the name of the parameter `headers_len` tells what is returned. So in this respect the function/parameters tell what is going on.
We can certainly bikeshed this more if you have a good proposal for the name(s). The `data_offset` is imho not a good name here as does not really give immediately the proper information. The question the reader might have is "offset to what?" if the name is `data_offset`.

Reply:
>The data_offset is imho not a good name here as does not really give immediately the proper information. The question the reader might have is "offset to what?" if the name is data_offset.

It would be "offset to data", similarly as function result is "length of data". So both properties (offset and length) would be to the same object, which is "data". In that case single function to return both (as a function result or by filling passed pointer) properties of the same "object" would make sense.